api/v1alpha1/composable_webhook.go

```diff
@@ -0,0 +1,271 @@
+		switch vv := v.(type) {
+		case string, int32, int64, float32, float64, bool:
+		case []interface{}:
+			if err := r.findGetValueFrom2(mykey, vv); err != nil {
+				allErrs = append(allErrs, err...)
+			}
+		case map[string]interface{}:
+			if vv["getValueFrom"] != nil {
+				newM, _ := json.Marshal(vv["getValueFrom"])
+				myGetValueFrom := ComposableGetValueFrom{}
+				json.Unmarshal(newM, &myGetValueFrom)
+				composablelog.Info("found getValueFrom", "key", "getValueFrom", "value", vv["getValueFrom"])
+				if err := validateGetValueFrom(mykey.String(), myGetValueFrom); err != nil {
+					composablelog.Info("getValueFrom is INVALID", "detail", err)
+					allErrs = append(allErrs, field.Invalid(mykey.Child("getValueFrom"), r.Name, err.Error()))
+				}
+				composablelog.Info("getValueFrom is valid")
+				//set to a random value for dry-run
+				// TODO: set the value to an appropriate type e.g. int, string, etc
+				m[k] = "abc"
+
+			} else { //recursive checking the sub-elements
+				if err := r.findGetValueFrom(mykey, vv); err != nil {
+					allErrs = append(allErrs, err...)
+				}
+			}
+		default:
+			composablelog.Info("findGetValueFrom", "key", mykey, "type-unknown", vv)
+		}
+	}
+	return allErrs
+}
+
+// findGetValueFrom2 functions in the same way as findGetValueFrom above except taking []interface{} as inputs
+// findGetValueFrom2 processes the ararys in the template content for getValueFrom elements
+func (r *Composable) findGetValueFrom2(path *field.Path, m []interface{}) field.ErrorList {
+	var allErrs field.ErrorList
+	for k, v := range m {
+		mykey := path.Child(strconv.Itoa(k))
+		switch vv := v.(type) {
+		case string, int32, int64, float32, float64, bool:
+		case []interface{}:
+			if err := r.findGetValueFrom2(mykey, vv); err != nil {
+				allErrs = append(allErrs, err...)
+			}
+		case map[string]interface{}:
+			if vv["getValueFrom"] != nil {
+				newM, _ := json.Marshal(vv["getValueFrom"])
+				myGetValueFrom := ComposableGetValueFrom{}
+				json.Unmarshal(newM, &myGetValueFrom)
+				composablelog.Info("found getValueFrom", "key", "getValueFrom", "value", vv["getValueFrom"])
+				if err := validateGetValueFrom(mykey.String(), myGetValueFrom); err != nil {
+					composablelog.Info("getValueFrom is INVALID", "detail", err)
+					allErrs = append(allErrs, field.Invalid(mykey.Child("getValueFrom"), r.Name, err.Error()))
+				}
+				composablelog.Info("getValueFrom is valid")
+				// TODO: set a random value of appropriate type for dry-run
+				m[k] = "abc2"
+			} else {
+				if err := r.findGetValueFrom(mykey, vv); err != nil {
+					allErrs = append(allErrs, err...)
+				}
+			}
+		default:
+			composablelog.Info("findGetValueFrom", "key", k, "type-unknown", vv)
+		}
+	}
+	return allErrs
+}
+
+// validateGetValueFrom validates the syntax of input getValueFrom fields
+func validateGetValueFrom(k string, getValueFrom ComposableGetValueFrom) error {
+	var missingItems []string
+	if getValueFrom.Name == "" {
```

Comment:
one of kuberntes code recommendations was to check that string is empty by `len(str) > 0` or  `len(str) == 0`

Reply:
good suggestion, i will make a change.

Reply:
the change has been checked in.